src/selective.d

```diff
@@ -194,23 +197,46 @@ final class SelectiveSync
 // if there are no allowed paths always return false
 private bool isPathExcluded(string path, string[] allowedPaths)
 {
+	string wildcard = "*";
 	// always allow the root
 	if (path == ".") return false;
 	// if there are no allowed paths always return false
 	if (allowedPaths.empty) return false;
-
 	path = buildNormalizedPath(path);
+	
 	foreach (allowed; allowedPaths) {
 		auto comm = commonPrefix(path, allowed);
+		// What are we comparing against?
+		log.vdebug("Evaluation against 'sync_list' entry: ", allowed);
+		// is path is an exact match
 		if (comm.length == path.length) {
 			// the given path is contained in an allowed path
+			log.vdebug("Evaluation against 'sync_list' result - direct match");
 			return false;
 		}
+		// is path is a subitem
 		if (comm.length == allowed.length && path[comm.length] == '/') {
 			// the given path is a subitem of an allowed path
+			log.vdebug("Evaluation against 'sync_list' result - parental path match");
 			return false;
 		}
+		
+		// does the allowed path contain a wildcard? (*)
+		if (canFind(allowed, wildcard)) {
+			// allowed path contains a wildcard
+			// manually escape '/'
+			string escapedAllowed = replace(allowed, "/", "\\/");
```

Comment:
Why is this necessary? `/` doesn't have a special meaning in regular expressions.
And what about `\*` as part of the file name? (but ok, that we can ignore - file names with literal `*` in it are a bit a pain)

Reply:
It needs to be escaped, so '/' in the allowed path, needs to be escaped with '\/' when passed in as a regex, thus needs to be '\\/' so that the regex can be passed correctly.

I wish I had all the development debug logs in the console, but from https://regex101.com/ - see below:

![regex1](https://user-images.githubusercontent.com/4956234/93302016-9f25c400-f83c-11ea-8331-d8794ac5eede.JPG)


![regex2](https://user-images.githubusercontent.com/4956234/93302028-a351e180-f83c-11ea-93d6-e34f76dd58f5.JPG)


![regex3](https://user-images.githubusercontent.com/4956234/93302033-a5b43b80-f83c-11ea-9cef-0329f262a48a.JPG)

Reply:
Well, that is what the `sed` requires because there 's/needle/replacement/options' is the normal syntax. But I don't see that D is using this. Have you actually tried without escapting the `/`? If that is necessary, the documentation of D regex library needs to be improved ;-)

Reply:
Anyway, this is in fact irrelevant, escaping is never a problem, so I approve.

Reply:
> Well, that is what the `sed` requires because there 's/needle/replacement/options' is the normal syntax. But I don't see that D is using this. Have you actually tried without escapting the `/`? If that is necessary, the documentation of D regex library needs to be improved ;-)

Yep .. the documentation needs to be improved .. the clue for me was here: https://dlang.org/phobos/std_regex.html#escaper

Reply:
Hmm that doesn't say anything about `/` though?

Reply:
In testing, I tested using 'escaper' to see what it would provide, and based on the output, use that to manually manipulate the input path to ensure that there was no breakage.

I have tested this further, and it appears that the escaping `/` is not needed:
```
Evaluation against 'sync_list' for: random_images/5vuzmRS6M6QsyXEGxze8oCoIxZAciXYN/image9.png
Evaluation against 'sync_list' entry: random_files/temp*
escapedString via escaper(allowed): random_files\/temp\*
escapedAllowed via manual replace: random_files/temp.*
Evaluation against 'sync_list' entry: random_ima*es
escapedString via escaper(allowed): random_ima\*es
escapedAllowed via manual replace: random_ima.*es
Evaluation against 'sync_list' result - wildcard pattern match
```
![regex4](https://user-images.githubusercontent.com/4956234/93387599-d764eb00-f8ac-11ea-92bc-9a0134afb15a.JPG)

Will fix this up in the PR I am developing for #1039

Reply:
This is what I expected. I don't know escaper, but regex ;-) Thanks, and as I said, it does not hurt so it is not a problem anyway.